frontend/src/container/RoutingPolicies/useRoutingPolicies.ts

```diff
@@ -0,0 +1,227 @@
+import './styles.scss';
+
+import { toast } from '@signozhq/sonner';
+import getAllChannels from 'api/channels/getAll';
+import { REACT_QUERY_KEY } from 'constants/reactQueryKeys';
+import { useCreateRoutingPolicy } from 'hooks/routingPolicies/useCreateRoutingPolicy';
+import { useDeleteRoutingPolicy } from 'hooks/routingPolicies/useDeleteRoutingPolicy';
+import { useGetRoutingPolicies } from 'hooks/routingPolicies/useGetRoutingPolicies';
+import { useUpdateRoutingPolicy } from 'hooks/routingPolicies/useUpdateRoutingPolicy';
+import { useMemo, useState } from 'react';
+import { useQuery, useQueryClient } from 'react-query';
+import { SuccessResponseV2 } from 'types/api';
+import { Channels } from 'types/api/channels/getAll';
+import APIError from 'types/api/error';
+
+import {
+	PolicyDetailsModalMode,
+	PolicyDetailsModalState,
+	RoutingPolicy,
+	UseRoutingPoliciesReturn,
+} from './types';
+import {
+	mapApiResponseToRoutingPolicies,
+	mapRoutingPolicyToCreateApiPayload,
+	mapRoutingPolicyToUpdateApiPayload,
+} from './utils';
+
+function useRoutingPolicies(): UseRoutingPoliciesReturn {
+	const queryClient = useQueryClient();
+
+	// Local state
+	const [searchTerm, setSearchTerm] = useState('');
+	const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+	const [
+		policyDetailsModalState,
+		setPolicyDetailsModalState,
+	] = useState<PolicyDetailsModalState>({
+		mode: null,
+		isOpen: false,
+	});
+	const [
+		selectedRoutingPolicy,
+		setSelectedRoutingPolicy,
+	] = useState<RoutingPolicy | null>(null);
+
+	// Routing Policies list
+	const {
+		data: routingPolicies,
+		isLoading: isLoadingRoutingPolicies,
+		isError: isErrorRoutingPolicies,
+	} = useGetRoutingPolicies();
+
+	const routingPoliciesData = useMemo(() => {
+		const unfilteredRoutingPolicies = mapApiResponseToRoutingPolicies(
+			routingPolicies,
+		);
+		return unfilteredRoutingPolicies.filter((routingPolicy) =>
+			routingPolicy.name.toLowerCase().includes(searchTerm.toLowerCase()),
+		);
+	}, [routingPolicies, searchTerm]);
+
+	// Channels list
+	const { data, isLoading: isLoadingChannels } = useQuery<
+		SuccessResponseV2<Channels[]>,
+		APIError
+	>(['getChannels'], {
+		queryFn: () => getAllChannels(),
+	});
+	const channels = data?.data || [];
```

Comment:
We need to handle this better. On error we should show error and on empty list, we should show message similar to what we do in alert create i.e ask them to create channel first (if they are not admin, ask them to get them created by admin)